jmbitcoin/jmbitcoin/secp256k1_main.py

```diff
@@ -137,24 +267,28 @@ def bin_sha256(string):
 def sha256(string):
     return bytes_to_hex_string(bin_sha256(string))
 
-def bin_dbl_sha256(s):
-    bytes_to_hash = from_string_to_bytes(s)
+def bin_dbl_sha256(bytes_to_hash):
     return hashlib.sha256(hashlib.sha256(bytes_to_hash).digest()).digest()
 
 def dbl_sha256(string):
-    return safe_hexlify(bin_dbl_sha256(string))
+    return hashlib.sha256(hashlib.sha256(string).digest()).hexdigest()
 
 def hash_to_int(x):
     if len(x) in [40, 64]:
         return decode(x, 16)
     return decode(x, 256)
 
 def num_to_var_int(x):
-    x = int(x)
+    if not isinstance(x, int):
+        if len(x) == 0:
+            return b'\x00'
+        x = struct.unpack('B', x)[0]
+    if x == 0:
+        return b'\x00'
```

Comment:
Not sure if this behavior is correct.

Reply:
This will fail in weird ways in py2 when passing a long. Why is the `isinstance` check needed?

Reply:
AFAIK `int` is remapped to long when using the future library.

Reply:
Yes, tried it, long(x) or 999L or similar, seems to work fine up to the range of varint (64 bit)

Reply:
If a string of length 0 is passed in, is the correct behavior to return `b'\x00'`?